arrow/src/main/java/org/apache/calcite/adapter/arrow/ArrowTranslator.java

```diff
@@ -114,7 +114,13 @@ private List<String> translateAnd(RexNode condition) {
     return predicates;
   }
 
-  /** Translate a binary or unary relation. */
+  /**
+   * Translates a binary or unary relation.
+   *
+   * @param node A RexNode that always evaluates to a boolean expression.
+   *             Currently, this method is only called from translateAnd.
```

Comment:
Since the current Arrow adapter does not support CNF, this method is currently only called from translateAnd

Reply:
A Boolean formula does not have to be in CNF or DNF form.
Currently Arrow supports only DNF forms for Boolean formulas.
There is an algorithm to convert any Boolean formula to DNF, but it can be very expensive in the worst case. There is no workaround, since the worst-case DNF formula can be exponential in the size of the original formula.

Reply:
Yes, I'm trying, but it seems very complicated

Reply:
Thank you very much for your advice, I am supporting CNF, but it seems that my thinking is wrong.